### PR TITLE
BUG Tweak the Eagerload test to not worry about sort order

### PR DESCRIPTION
**Pull request by maxime-rainville**
PostgreSQL returns results in a different order than MySQL.

## Depends on
- https://github.com/silverstripe/silverstripe-postgresql/pull/148

## Parent issue
- https://github.com/silverstripe/silverstripe-postgresql/issues/149

### CODE REVIEW

File: tests/php/ORM/DataListTest.php [hunk 1]
```diff
@@ -58,6 +58,8 @@ class DataListTest extends SapphireTest
     // Borrow the model from DataObjectTest
     protected static $fixture_file = 'DataObjectTest.yml';
 
+    protected $use_database = true;
```
Comment (GuySartorelli) [inline tests/php/ORM/DataListTest.php]: Why is this line added? It's not needed. "Note that this flag is overruled by the presence of a `$fixture_file`, which always forces a database build."

File: tests/php/ORM/DataListTest.php [hunk 2]
```diff
@@ -2518,6 +2498,25 @@ private function createEagerLoadData(): void
...
+                    // $results = DB::get_schema()->preparedQuery(
+                    //     "SELECT table_name, column_name, data_type FROM information_schema.columns WHERE data_type='tsvector' AND table_name like 'ManyManyThroughEagerLoadObjectManyManyThroughSubEagerLoadObject';", []
+                    // );
+
+                    // var_dump(DB::get_schema()->fieldList('ManyManyThroughEagerLoadObjectManyManyThroughSubEagerLoadObject'));
+
+                    // foreach ($results as $result) {
+                    //     var_dump($result);
+                    // }
+
```
Comment (GuySartorelli) [inline tests/php/ORM/DataListTest.php]: We can't just have a bunch of commented out cruft in here 😅

File: tests/php/ORM/DataListTest.php [hunk 3]
```diff
@@ -2254,172 +2256,150 @@ private function chunkTester(
...
+    public function testEagerLoadRelations(array $eagerLoad, int $expectedCount): void
     {
         $this->createEagerLoadData();
         $dataList = EagerLoadObject::get()->eagerLoad(...$eagerLoad);
         list($results, $selectCount) = $this->iterateEagerLoadData($dataList);
-        $this->assertSame($this->expectedEagerLoadData(), $results, $iden);
-        $this->assertSame($expected, $selectCount, $iden);
+        $expectedResults=$this->expectedEagerLoadData();
+        $this->assertSame(sort($expectedResults), sort($results));
+        $this->assertSame($expectedCount, $selectCount);
```
Comment (GuySartorelli) [inline tests/php/ORM/DataListTest.php:2264]: It seems like this should be all that's needed.... what are all the other changes for? (note that I haven't taken much time to try to understand it - I'm assuming you explaining it will be faster as I've not looked at this test code at all and it seems there's quite a lot going on)

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: tests/php/ORM/DataListTest.php (hunk 1 above)
Comment (maxime-rainville) [reply tests/php/ORM/DataListTest.php]: Forgot to remove a bunch of rando debug code.

File: tests/php/ORM/DataListTest.php (hunk 2 above)
Comment (maxime-rainville) [reply tests/php/ORM/DataListTest.php]: Removed.

File: tests/php/ORM/DataListTest.php (hunk 3 above)
Comment (maxime-rainville) [reply tests/php/ORM/DataListTest.php:2264]: I think using the dataprovider array key to identify the test cases is a generally better syntax so I'm updated those.

But I'm not fuss either way. I can revert that back if it's a problem.

Comment (GuySartorelli) [reply tests/php/ORM/DataListTest.php:2264]: If you want to keep those changes, please separate them into a _separate commit_ in this PR so I can review them independently of the other changes.

**Comment by GuySartorelli** [comment]
Please also change the commit message prefix from `BUG` to `MNT` as per https://docs.silverstripe.org/en/5/contributing/code/#commit-messages

File: tests/php/ORM/DataListTest.php (hunk 3 above)
Comment (maxime-rainville) [reply tests/php/ORM/DataListTest.php:2264]: Change has been split into two commits.

**Review by GuySartorelli — APPROVED** [review]
Look okay to me.
The unrelated change is very subjective - on the one hand that's similar to how a lot of our other unit tests have their providers. On the other hand, this makes it harder to see _which_ case failed, if one does fail.

I'm not really fussed either way and I can see that the change doesn't break or weaken the test itself so I'm okay merging it as is.

**Comment by maxime-rainville** [comment]
Let's maybe leave it open until we know for sure the build is green on this module. https://github.com/silverstripe/silverstripe-postgresql/actions/runs/5435897366
